src/pipeline.ts

```diff
@@ -175,8 +214,16 @@ export class GitHubWorkflow extends PipelineBase {
   private readonly assetHashMap: Record<string, string> = {};
   private readonly runner: github.Runner;
   private readonly publishAssetsAuthRegion: string;
-  private readonly stackProperties: Record<string, Record<string, any>> = {};
+  private readonly stackProperties: Record<
+  string,
+  {
+    environment: AddGitHubStageOptions['gitHubEnvironment'];
+    capabilities: AddGitHubStageOptions['stackCapabilities'];
+    settings: AddGitHubStageOptions['jobSettings'];
+  }
+  > = {};
   private readonly jobSettings?: JobSettings;
+  private builtGH = false; // cannot be `built` since that's defined in the parent as private
```

Comment:
The parent does this check [here](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/pipelines/lib/main/pipeline-base.ts#L115-L117) in `PipelineBase.addWave()`. We can't use `this.built` since it's declared as `private` in `PipelineBase`, and we need to block the user from adding a Wave after the pipeline has built.

We can't use the provided `PipelineBase.addWave()` since it'll make a new `Wave`, so we override it.

I'm adding the comment:
```ts
  // in order to keep track of if this pipeline has been built so we can
  // catch later calls to addWave() or addStage()
```